test/test_retriever.py

```diff
@@ -691,3 +692,17 @@ def test_setup_functions():
     assert os.path.exists(raw_dir_files.format(file_name="")) is False
     setup_functions()
     assert os.path.exists(raw_dir_files.format(file_name=""))
+
+
+def test_reset_retriever(tmpdir):
+    """Test the dataset reset function."""
+
+    pwd_name = os.getcwd()
+    workdir = tmpdir.mkdtemp()
+    workdir.chdir()
+    dataset = random.choice(rt.dataset_names())
+    rt.reset_retriever(dataset)
+    rt.reload_scripts()
+    assert dataset not in rt.dataset_names()
+    os.chdir(pwd_name)
+    rt.check_for_updates()
```

Comment:
```python
 rt.reload_scripts()
 assert dataset in rt.dataset_names()
```
@harshitbansal05 we can also add the next check here to test that we have up dated.
I guess the check should also be in tmp dir.